starknet-rpc-test/get_block_transaction_count.rs

```diff
@@ -0,0 +1,88 @@
+extern crate starknet_rpc_test;
+
+use starknet_accounts::SingleOwnerAccount;
+use starknet_core::chain_id;
+use starknet_core::types::{BlockId, BlockTag};
+use starknet_ff::FieldElement;
+use starknet_providers::Provider;
+use starknet_rpc_test::constants::{ARGENT_CONTRACT_ADDRESS, MINT_AMOUNT, SIGNER_PRIVATE};
+use starknet_rpc_test::utils::transfer_tokens;
+use starknet_rpc_test::{BlockCreation, ExecutionStrategy, MadaraClient};
+use starknet_signers::{LocalWallet, SigningKey};
+use tokio::sync::OnceCell;
+
+static MADARA_CLIENT: OnceCell<MadaraClient> = OnceCell::const_new();
+
+async fn get_madara_client() -> &'static MadaraClient {
+    MADARA_CLIENT.get_or_init(|| async { MadaraClient::new(ExecutionStrategy::Native).await }).await
+}
+
+#[tokio::test]
+async fn work_ok_with_empty_block() -> Result<(), anyhow::Error> {
+    let madara = get_madara_client().await;
+    let rpc = madara.get_starknet_client();
+
+    assert_eq!(rpc.get_block_transaction_count(BlockId::Tag(BlockTag::Latest)).await?, 0);
+
+    madara.create_block(vec![], BlockCreation::default()).await?;
```

Comment:
I think the caller shouldn't have to worry about the parameters he calls this method with. Those are details we should abstract.

I would rather have multiple methods

```rust
madara.create_empty_block();
madara.create_block_with_txs(vec![<my txs>]);
madara.create_block_with_parent("0xbabe".to_string());
```

For now, there is no use case for creating an unfinalized block (correct me if I'm wrong), so just don't worry about it, set it to `true` every time without consulting the caller.